exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -60,16 +61,227 @@ const (
 	identifierOffset = 11 // offset of identifier within traceID
 )
 
+const (
+	localRoot = "LOCAL_ROOT"
+)
+
+var removeAnnotationsFromServiceSegment = []string{
+	"aws.remote.service",
+	"aws.remote.operation",
+	"remote.target",
+	"K8s.RemoteNamespace",
+}
+
 var (
 	writers = newWriterPool(2048)
 )
 
-// MakeSegmentDocumentString converts an OpenTelemetry Span to an X-Ray Segment and then serialzies to JSON
+// MakeSegmentDocuments converts spans to json documents
+func MakeSegmentDocuments(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool) ([]string, error) {
+	segments, err := MakeSegmentsFromSpan(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err == nil {
+		var documents []string
+
+		for _, v := range segments {
+			document, documentErr := MakeDocumentFromSegment(v)
+			if documentErr != nil {
+				return nil, documentErr
+			}
+
+			documents = append(documents, document)
+		}
+
+		return documents, nil
+	}
+
+	return nil, err
+}
+
+func HasDependencySubsegment(span ptrace.Span) bool {
+	_, hasAwsRemoteService := span.Attributes().Get(awsRemoteService)
+
+	return span.Kind() != ptrace.SpanKindServer &&
+		span.Kind() != ptrace.SpanKindInternal &&
+		hasAwsRemoteService
```

Comment:
Sorry for the confusion, but similar to my comment on `MakeServiceSegmentWithoutDependency` this does not actually care about `hasAwsRemoteService` because we are in the `LocalRoot` workflow. We can remove `hasAwsRemoteService` logic to keep things simple. 

However, I think we can should rename to `isLocalRootSpanADependencySpan` to make it explicit what we are working with, to avoid confusion.

Similar rename suggestion for `MakeDependencySubsegment` -> `MakeDependencySubsegmentForLocalRootDependencySpan` and `MakeServiceSegment` -> `MakeServiceSegmentForLocalRootDependencySpan`